### PR TITLE
Improve IO module typing

### PR DESCRIPTION
**Pull request by felixdivo**
This also fixes the mypy issue found in #1061.

### CODE REVIEW

**Comment by felixdivo** [comment]
@Mergifyio rebase

File: can/io/generic.py [hunk 1]
```diff
@@ -3,19 +3,20 @@
 """
 
 from abc import ABCMeta
-from typing import Any, Optional, cast, Union, TextIO, BinaryIO, Type
+from typing import Any, Optional, cast, Iterable, Union, TextIO, BinaryIO, Type, ContextManager
+from typing_extensions import Literal
```
Comment (zariiii9003) [inline can/io/generic.py:7]: We do not have the `typing_extensions` dependency in setup.py. Do we need it?

Comment (felixdivo) [reply can/io/generic.py:7]: Yeah, also realized that. It somehow was present in my venv though so I thought I'd give it a try.

I'd propose to add it, yes. It's the backport of newer typing additions to older Python versions (the `Literal` was [added in 3.8](https://docs.python.org/3/library/typing.html#typing.Literal)). Shall I add it to this PR?

Comment (zariiii9003) [reply can/io/generic.py:7]: Personally i don't mind it.

**Comment by felixdivo** [comment]
@zariiii9003 It should work now.

**Comment by zariiii9003** [comment]
Mypy still does not seem happy: https://travis-ci.com/github/hardbyte/python-can/jobs/511375046#L586

**Comment by felixdivo** [comment]
@zariiii9003 It seems like pylint does not recognize the type annotations that it really is an `Iterable` and reports a false positive. Everything work now. Do you approve this PR?

**Review by zariiii9003 — APPROVED** [review]

**Pull request merged**